### PR TITLE
Fixes bad default ethereal preference color

### PR DESCRIPTION
**Pull request by ShizCalev**
Fixed malformed hex runtime since the `#` is automatically prefixed to this value when it's present.

Think this is the right way, but I hate touching savefile crap. @Cyberboss @MrStonedOne look good to you guys?

```
[18:31:07] Runtime in type2type.dm, line 34: Malformed hex number
proc name: hex2num (/proc/hex2num)
usr: BudyMan1/(BudyMan1)
usr.loc: (start area (8, 174, 1))
src: null
call stack:
hex2num("#9", 0)
GetRedPart("##9c303")
Ethereal (/datum/species/ethereal): on species gain(Ivan Andropow (/mob/living/carbon/human/dummy), Flyperson (/datum/species/fly), 1)
Ivan Andropow (/mob/living/carbon/human/dummy): set species(/datum/species/ethereal (/datum/species/ethereal), 0, 1)
Ivan Andropow (/mob/living/carbon/human/dummy): set species(/datum/species/ethereal (/datum/species/ethereal), 0, 1)
/datum/preferences (/datum/preferences): copy to(Ivan Andropow (/mob/living/carbon/human/dummy), 1, 1)
/datum/preferences (/datum/preferences): update preview icon()
/datum/preferences (/datum/preferences): ShowChoices(BudyMan1 (/mob/dead/new_player))
/datum/preferences (/datum/preferences): process link(BudyMan1 (/mob/dead/new_player), /list (/list))
BudyMan1 (/client): Topic("_src_=prefs;preference=species...", /list (/list), null)
BudyMan1 (/client): Topic("_src_=prefs;preference=species...", /list (/list))
```

:cl: ShizCalev
fix: Fixed incorrect default ethereal color causing a runtime when opening player preferences.
/:cl:

### CODE REVIEW

**Comment by optimumtact** [comment]
Ah, I chased this for a while but floyd was being evasive when I bailed him up about it

LGTM

**Review by Cyberboss — DISMISSED** [review]

File: code/modules/client/preferences_savefile.dm [hunk 1]
```diff
@@ -49,6 +49,9 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 		pda_style = "mono"
 	if(current_version < 20)
 		pda_color = "#808000"
+	if((current_version < 21) && S["feature_ethcolor"] && (S["feature_ethcolor"] == "#9c3030"))
+		WRITE_FILE(S["feature_ethcolor"]	, "9c3030")
```
Comment (MrStonedOne) [inline code/modules/client/preferences_savefile.dm]: This proc is called after the character file is loaded into the pref datum, so you should be reading and changing the datum vars not the savefile values.

Also the system should save the save file when it does the version conversion, it's not currently doing that. that can be fix in another pr.
